embc-app/ClientApp/src/app/volunteer/components/valid-from-to/valid-from-to.component.ts

```diff
@@ -1,19 +1,157 @@
-import { Component, OnInit, Input } from '@angular/core';
+import { Component, EventEmitter, Input, Output, OnInit } from '@angular/core';
 import range from 'lodash/range';
+import { ReferralDate, ReferralDateForm, YearMonthDay, HourMinute } from 'src/app/core/models/referral-date';
+import * as moment from 'moment';
 
 @Component({
   selector: 'app-valid-from-to',
   templateUrl: './valid-from-to.component.html',
   styleUrls: ['./valid-from-to.component.scss']
 })
 export class ValidFromToComponent implements OnInit {
-  @Input() validFrom: Date = null;
-  @Input() validTo: Date = null;
+  @Input() editMode: boolean; // unimplemented editmode false is read only
+  @Input() referralDate: ReferralDate;
+  @Input() id?: string;
+  @Output() date = new EventEmitter<ReferralDate>();
 
   days = range(1, 15); // [1..14]
+  defaultDays = 1; // the default amount for the component to use as a duration
+  displayDate: string;
+  displayTime: string;
+  // invalid date or time
+  validDate = true; // begin with the assumption of validity
+  validTime = true;
+
+  wrdForm: ReferralDateForm; // The form elements (different data)
 
   constructor() { }
 
-  ngOnInit() { }
+  ngOnInit() {
+    // if the id is missing we set the component to generic
+    if (!this.id) {
+      this.id = 'generic';
```

Comment:
FYI, I think this could be simplified by just initializing id in the @Input declaration.

Reply:
Ok. I'll add that.